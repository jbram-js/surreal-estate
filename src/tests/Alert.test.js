import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("displays error message", () => {
    const { getByText, asFragment } = render(<Alert message="Error!" />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Error/).textContent).toBe("Error!");
  });

  it("displays success message", () => {
    const { getByText, asFragment } = render(
      <Alert message="Success!!!" success />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Success/).textContent).toBe("Success!!!");
  });

  it("does not render is message is not truthy", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
    expect(asFragment()).toMatchSnapshot();
  });
});
