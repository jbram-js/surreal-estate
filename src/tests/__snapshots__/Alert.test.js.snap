// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays error message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert displays success message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-success"
  >
    Success!!!
  </div>
</DocumentFragment>
`;

exports[`Alert displays success message 2`] = `
<DocumentFragment>
  <div
    class="Alert alert-success"
  >
    Success!!!
  </div>
</DocumentFragment>
`;

exports[`Alert does not render is message is not truthy 1`] = `<DocumentFragment />`;

exports[`Alert does not render is message is not truthy 2`] = `<DocumentFragment />`;

exports[`displays a success message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-success"
  >
    Property Added
  </div>
</DocumentFragment>
`;

exports[`displays an error message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`does not render an error or a success message if message props is empty 1`] = `<DocumentFragment />`;
